File: felicia/examples/learn/common/cc/node_create_flag.cc
#include "felicia/examples/learn/common/cc/node_create_flag.h"

namespace felicia {

NodeCreateFlag::NodeCreateFlag() {
  {
    StringFlag::Builder builder(MakeValueStore(&name_));
    auto flag = builder.SetShortName("-n")
                    .SetLongName("--name")
                    .SetHelp("name for node")
                    .Build();
    name_flag_ = std::make_unique<StringFlag>(flag);
  }
}

NodeCreateFlag::~NodeCreateFlag() = default;

bool NodeCreateFlag::Parse(FlagParser& parser) {
  return PARSE_OPTIONAL_FLAG(parser, name_flag_);
}

}  // namespace felicia

File: felicia/examples/learn/common/python/node_create_flag.py
from felicia.python.command_line_interface.flag_parser_delegate import FlagParserDelegate
import felicia_py.command_line_interface as cli


class NodeCreateFlag(FlagParserDelegate):
    def __init__(self):
        super().__init__()
        self.name_flag = cli.StringFlagBuilder().set_short_name(
            "-n").set_long_name("--name").set_help("name for node").build()

    def parse(self, flag_parser):
        return self.parse_optional_flags(flag_parser)
